dotcom-rendering/src/web/lib/hasAds.ts

```diff
@@ -2,13 +2,18 @@ import type { DCRFrontType } from '../../types/front';
 import type { FEArticleType } from '../../types/frontend';
 
 export const hasAds = (pageData: FEArticleType | DCRFrontType): boolean => {
-	let render =
-		!pageData.isAdFreeUser &&
-		pageData.config.abTests.poorDeviceConnectivityVariant !== 'variant';
+	if (pageData.isAdFreeUser) {
+		return false;
+	}
 
+	// DCRFrontType doesn't have a shouldHideAds property
 	if ('shouldHideAds' in pageData && pageData.shouldHideAds) {
```

Comment:
Is this required [because of the safer comparison linting rule](https://github.com/guardian/dotcom-rendering/pull/7152)? If so, I think `pageData.shouldHideAds === true` is a little more explicit.

Reply:
yeah i guess so, it's complaining it does not exist on `DCRFrontType`. but it can still only be a boolean. if testing for `true` is more explicit shouldn't we require it everywhere and lint it?

Reply:
[This is exactly what we do](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md#-correct). Maybe the linter is not explicit enough, but it’s preventing you from having an ambiguous value in your if statement (`boolean | undefined`) and wants a strict `boolean`.

```ts
if(pageData.shoulHideAds) {
  /* oh no */
}

if(pageData.shoulHideAds === true) {
  /* happy linter, happy days */
}
```

Reply:
<img width="728" alt="image" src="https://user-images.githubusercontent.com/867233/225054942-39f44ca5-8d75-4ad1-a78d-3caafa3a7cf3.png">

Reply:
it TS, not the linter